core/src/main/java/org/apache/iceberg/BaseFile.java

```diff
@@ -309,17 +306,17 @@ public Object get(int i) {
       case 5:
         return fileSizeInBytes;
       case 6:
-        return columnSizes;
+        return toReadableMap(columnSizes);
       case 7:
-        return valueCounts;
+        return toReadableMap(valueCounts);
       case 8:
-        return nullValueCounts;
+        return toReadableMap(nullValueCounts);
       case 9:
-        return nanValueCounts;
+        return toReadableMap(nanValueCounts);
       case 10:
-        return lowerBounds;
+        return toReadableMap(lowerBounds);
       case 11:
-        return upperBounds;
+        return toReadableMap(upperBounds);
```

Comment:
I don't think we need to use `toReadableMap` in this method because it is only used for serialization. We don't expose the Avro methods publicly, so we only need the unmodifiable maps to be returned by the `ContentFile` accessor methods.

Reply:
OK, that make sense !